src/components/SaveIndicator.tsx

```diff
@@ -0,0 +1,31 @@
+import { useIntl } from "react-intl";
+import { SaveStatus } from "../abstractions/common/save-status";
+
+export function SaveIndicator({ saveStatus }: { saveStatus: SaveStatus }) {
+  const intl = useIntl();
+
+  const { statusClassName, statusText, statusDescription } =
+    saveStatus === "idle"
+      ? {
+          statusClassName: "save-indicator-idle",
+          statusText: intl.formatMessage({ id: "saved" }),
+          statusDescription: intl.formatMessage({ id: "saved_details" }),
+        }
+      : saveStatus === "saved"
+      ? {
+          statusClassName: "save-indicator-saved",
+          statusText: intl.formatMessage({ id: "saved" }),
+          statusDescription: intl.formatMessage({ id: "saved_details" }),
+        }
+      : {
+          statusClassName: "save-indicator-saving",
```

Comment:
Actually, we have more identified states:

* `pending` when there are pending changes but we did not start the saving process yet.
* `saving` when it is actually saving (and we could split it in two: `preparing` and `saving`)

In my opinion, from the user's perspective, all of them are the same, but if you want, we could define classes for each one in case we want different styles.